ui/packages/catalog-ui-search/src/main/webapp/component/query-template-sharing/query-template-sharing.less

```diff
@@ -0,0 +1,111 @@
+@import '../editable-rows/editable-rows';
+
+.icon-label {
+  padding-left: @minimumSpacing;
+}
+
+@{customElementNamespace}template-sharing {
+
+  @valueWidth: 60%;
+  position: relative;
+
+  .customElement;
+
+  .text-right {
+    text-align: right;
+  }
+
+  overflow: auto;
+
+  .sharing-buttons {
+    position: absolute;
+    bottom: 20px;
+    left: 20px;
+    width: ~'calc(100% - 40px)';
+    white-space: nowrap;
+
+    button {
+      width: 50%;
+      vertical-align: top;
+    }
+  }
+
+  .template-sharing-by-role {
+    .role {
+      line-height: @minimumButtonSize;
+      height: @minimumButtonSize;
+      text-align: center;
+    }
+    .row {
+      white-space: nowrap;
+      margin: 0px 0px @minimumSpacing 0px;
+      padding: 0px @minimumSpacing;
+    }
+
+    .role,
+    .action {
+      vertical-align: top;
+      display: inline-block;
+      width: 50%;
+    }
+  }
+
+  .template-sharing-by-email {
```

Comment:
I think in this case it makes more sense to leave this as-is for now since this class is bound to a specific div supporting email and only email. If we added a new feature, it would most likely exist within it's own div and have custom styling associated with that feature similar to roles. 

However, if there is a feature that replicates the same usage of editable rows in this case, it would be nice to rename to be reusable if they happened to be identical in functionality.